pkg/resource/stack/deployment.go

```diff
@@ -456,7 +456,7 @@ func DeserializeResource(res apitype.ResourceV3, dec config.Decrypter, enc confi
 		res.Type, res.URN, res.Custom, res.Delete, res.ID,
 		inputs, outputs, res.Parent, res.Protect, res.External, res.Dependencies, res.InitErrors, res.Provider,
 		res.PropertyDependencies, res.PendingReplacement, res.AdditionalSecretOutputs, res.Aliases, res.CustomTimeouts,
-		res.ImportID), nil
+		res.ImportID, nil /* TODO: Is it okay that this doesn't get deserialized? */), nil
```

Comment:
I'm fairly sure there is no value in serializing `ReplaceOnChanges` to the state file, much like we do not serialize `IgnoreChanges` to state.  `ReplaceOnChanges` is only ever read off the `new` state defined by a program.

Reply:
The only potential case I can see for saving that state is for audit purposes. I'm not sure if that information is captured by the update diff? If not, resources could get replaced from uncommitted local changes, and the cause would not be obvious.

Reply:
I've aligned the implementation here with `ignoreChanges` for now for consistency.  It is quite likely that we want both `ignoreChanges` and `replaceOnChanges` to be serialized into state though - and I've opened https://github.com/pulumi/pulumi/issues/7285 to track that as a follow-up - most likely needed as part of implementing rollback support for https://github.com/pulumi/pulumi/issues/96.